src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStore.java

```diff
@@ -185,18 +186,30 @@ public URL toExternalURL(@NonNull Blob blob, @NonNull HttpMethod httpMethod) thr
         String name = blob.getMetadata().getName();
         LOGGER.log(Level.FINE, "Generating presigned URL for {0} / {1} for method {2}",
                 new Object[] { container, name, httpMethod });
+        String contentType = null;
         com.amazonaws.HttpMethod awsMethod;
         switch (httpMethod) {
         case PUT:
             awsMethod = com.amazonaws.HttpMethod.PUT;
+            // Only set content type for upload URLs, so that the right S3 metadata gets set
+            contentType = blob.getMetadata().getContentMetadata().getContentType();
             break;
         case GET:
             awsMethod = com.amazonaws.HttpMethod.GET;
             break;
         default:
             throw new IOException("HTTP Method " + httpMethod + " not supported for S3");
         }
-        return builder.build().generatePresignedUrl(container, name, expiration, awsMethod);
+
+        GeneratePresignedUrlRequest generatePresignedUrlRequest = new GeneratePresignedUrlRequest(container, name)
+            .withExpiration(expiration)
+            .withMethod(awsMethod);
+
+        if (contentType != null) {
```

Comment:
GeneratePresignedUrlRequest.setContentType allows null. It might be easier on the eyes if the chain above just calls withContentType without bothering with the null check

Reply:
Will do!